llvm/test/CodeGen/AMDGPU/flat-atomicrmw-fmin.ll

```diff
@@ -31,13 +31,13 @@ define float @flat_agent_atomic_fmin_ret_f32__amdgpu_no_fine_grained_memory(ptr
 ; GFX940-NEXT:    s_waitcnt vmcnt(0) expcnt(0) lgkmcnt(0)
 ; GFX940-NEXT:    flat_load_dword v3, v[0:1]
 ; GFX940-NEXT:    s_mov_b64 s[0:1], 0
-; GFX940-NEXT:    v_max_f32_e32 v2, v2, v2
 ; GFX940-NEXT:  .LBB0_1: ; %atomicrmw.start
 ; GFX940-NEXT:    ; =>This Inner Loop Header: Depth=1
 ; GFX940-NEXT:    s_waitcnt vmcnt(0) lgkmcnt(0)
 ; GFX940-NEXT:    v_mov_b32_e32 v5, v3
+; GFX940-NEXT:    v_max_f32_e32 v4, v2, v2
```

Comment:
Most of these test changes look like no longer hoisting instructions out of atomic cmpxchg loops, and are probably undesirable. I wouldn't expect there to be this much of a change from this on any of these cases

Reply:
AMDGPU is the only target that has overrided `TargetRegisterInfo::getRegPressureSetLimit`, is it the reason? For AMDGPU, we shouldn't remove reserved registers?

Reply:
We don't see these changes in https://github.com/llvm/llvm-project/pull/118787, these instructions may be sinked if we also change MachineSink. So, I think this is not blocking.

Reply:
So, should I fix these changes in this patch? Or should I just go ahead?

Reply:
if you push the patches in the other order, do the transient regressions disappear?

Reply:
After #118787, all AMDGPU changes disappear. This is because AMDGPU overrided the `TargetRegisterInfo::getRegPressureSetLimit`.